### PR TITLE
fix: Limits throughput when reading from kafka

### PR DESCRIPTION
**Pull request by tisonet**

### CODE REVIEW

File: direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptorTest.java [hunk 1]
```diff
@@ -2609,7 +2609,7 @@ public void testRewriteAndPrefetch() throws InterruptedException, IOException {
     assertArrayEquals(new byte[] {3, 4}, view.get("key1", attr).get().getValue());
   }
 
-  @Test(timeout = 5000)
+  @Test(timeout = 50000)
```
[the remark this answers is not in the document]
Comment (tisonet) [reply direct/io-kafka/src/test/java/cz/o2/proxima/direct/kafka/LocalKafkaCommitLogDescriptorTest.java]: No, just changed for debug purpose, reverting.

File: direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaPollLimiter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+/** Limits poll throughput when reading from Kafka. */
+class KafkaPollLimiter {
+  private final long desiredBytesPerMs;
+
+  public KafkaPollLimiter(long maxBytesPerSec) {
+    this.desiredBytesPerMs =
+        maxBytesPerSec < Long.MAX_VALUE ? Math.max(1L, maxBytesPerSec / 1000) : Long.MAX_VALUE;
+  }
+
+  public void limitPoll(long bytesPolled, long poolTimeMs) throws InterruptedException {
```
[the remark this answers is not in the document]
Comment (tisonet) [reply direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaPollLimiter.java]: Will try

**Review by je-ik — APPROVED** [review]
👍

**Comment by je-ik** [comment]
Hm, this didn't go through CI. Travis-ci.org was not working anymore at the time we merged this.

**Comment by je-ik** [comment]
(and the build is broken, which is common, when it was not run :))
fix: https://github.com/datadrivencz/proxima-platform/pull/566

**Pull request merged**